inference-engine/tests/functional/plugin/cpu/single_layer_tests/adaptive_pooling.cpp

```diff
@@ -62,14 +71,14 @@ class AdaPoolLayerCPUTest : public testing::WithParamInterface<AdaPoolLayerCPUTe
         std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
 
         CPULayerTestsDefinitions::AdaPoolSpecificParams adaPoolParams;
-        auto netPrecision = InferenceEngine::Precision::UNSPECIFIED;
+        ElementType netPrecision;
         std::tie(adaPoolParams, mode,  netPrecision, targetDevice) = basicParamsSet;
-        inPrc = outPrc = netPrecision;
         std::tie(pooledSpatialShape, inputShape) = adaPoolParams;
 
         ngraph::Shape coordsShape = {pooledSpatialShape.size() };
         auto pooledParam = ngraph::builder::makeConstant<int32_t>(ngraph::element::i32, coordsShape, pooledSpatialShape);
-        auto params = ngraph::builder::makeParams(ngraph::element::f32, {inputShape});
+        init_input_shapes(inputShape);
```

Comment:
is it possibly to modify test to check cases where second input also has dynamic shapes?

for this you need to create `coordsShape` as parameter and override `generate_inputs` to fill second input

Reply:
Hi @mandrono  I got below error if set the second input as dynamic shape, because validate_and_infer_types() gets the second input from constant and infer the output shape with this constant value. If the second input is not constant, the get_constant_from_source() will return false and output shape of correspond dim will be set to partial shape.

Please let me know if you have any suggestions.

Error message:
openvino/inference-engine/src/mkldnn_plugin/mkldnn_node.cpp:1338 [ NOT_IMPLEMENTED ] CPU plug-in doesn't support default shape infer for nodes with internal dynamism

Reply:
Since output shape depends on input data you need to override `needShapeInfer` to check cases when input(0) shape is same but input(1) data was changed.
Also you need to override `shapeInfer` you can do it like that https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_one_hot_node.cpp#L86-L106
but shape infer for adaptive pooling look simple so better to reimplement it inside plug-in due to  performance

If after this changes problem with test will appear please use the same approach as here:
https://github.com/openvinotoolkit/openvino/blob/24675fd8ea11f964cd732eda1f27cdaa3d18dd20/inference-engine/tests/functional/plugin/cpu/single_layer_tests/one_hot.cpp#L105-L111

Reply:
Done.